hayeon/programmers-0/문자열-밀기.js

```diff
@@ -0,0 +1,15 @@
+function solution(A, B) {
+  if (A === B) return 0;
+
+  for (let i = 0; i < A.length; i++) {
+    A = A.slice(-1) + A.slice(0, -1);
+    if (A === B) return i + 1;
+  }
+  return -1;
+}
```

Comment:
```suggestion
function solution(A, B) {
  let extrudedStr = A;

  if (A === B) return 0;

  for (let i = 0; i < A.length; i++) {
    const lastChar = extrudedStr.slice(-1);
    const slicedStr = extrudedStr.slice(0, -1);

    extrudedStr = lastChar + slicedStr;

    if (extrudedStr === B) return i + 1;
  }

  return -1;
}
```
- slice를 이용한 풀이 잘봤습니다~ 상황마다 다르겠지만 웬만하면 매개변수를 직접 건드리는 것은 지양해야됩니다. extrudedStr와 같은 변수를 하나 만들어서 이를 핸들링하시는 것을 추천드립니다 !~ 🙇‍♂️
- 놀랍게도 아래와 같은 풀이도 가능합니다.
```js
function solution(A, B) {
  return (B + B).indexOf(A);
}
```